jv-array-list

OleksiiBashkirov opened the pull request:
created ArrayList
added checkCapacity method

olekskov commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,102 @@
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
```
Please extract repeated logic to separate method

olekskov commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,102 @@
     @Override
     public T remove(T element) {
-        return null;
+        if (element == null) {
+            for (int i = 0; i < size; i++) {
+                if (elements[i] == null) {
+                    return remove(i);
+                }
+            }
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (element.equals(elements[i])) {
+                    return remove(i);
+                }
+            }
+        }
```
Please compare null and not null objects in one if
for (int i = 0; i < size; i++) {
    if (element == elements[i] || element != null && element.equals(elements[i])) {
        // your logic
    }
}

OleksiiBashkirov commented:
extracted repetitive logic into the checkIndex method
also created checkIndexForAdd method 
modified public T remove(T element) method with a single if
